### DISCUSSION TITLE
How can I achieve Autocomplete Prediction Precedence?

### DISCUSSION THREAD

**Discussion by fzkhan19** [visgl/react-google-maps#493] [Q&A]
I want the autocomplete to give precedence to airports/terminals and that should be visible on top of the search results, without blocking off other results.

For eg. if I search for lemas bakery ajman it should show that result. but if I type dubai the top result should be of airport and terminals.

I know about types, and I've used it as well but then it blocks of other results.

This  is my code:
```
const request: google.maps.places.AutocompletionRequest = {
        input: inputValue,
        sessionToken,
        types: ["airport", "lodging", "point_of_interest", "food", "street_address"],
        locationBias: new google.maps.LatLngBounds(
          new google.maps.LatLng(latitude, longitude),
          new google.maps.LatLng(latitude + 0.5, longitude + 0.5),
        ),
        //Country Restricted to London

        componentRestrictions: {country: countryCode as string},
      };
      const response = await autocompleteService.getPlacePredictions(request);
```

**Comment by mrMetalWood** [comment]
I'm not aware of any "built-in" features that would allow such behaviour.

Maybe it could work by always making two autocomplete requests. One with only `airport` and an unrestricted one. Once there are not enough airports returned to fill your suggestion list, fill the list with the top results from the unrestricted suggestion call. 

Not sure how well this would work. Note that this will probably also impact billing, since you are always making two requests for every change.
